### DISCUSSION TITLE
Generated schema not the correct Type?

### DISCUSSION THREAD

**Discussion by GollyJer** [urql-graphql/urql#1321] [Q&A] [answered]
Hi. I'm setting up Schema Awareness and want to be sure I haven't messed up.
I followed the [Optimizing A Schema](https://formidable.com/open-source/urql/docs/graphcache/schema-awareness/#optimizing-a-schema) section exactly.

However, when I pass the generated schema to `cacheExchange` I get this type error. 

> <img src=https://user-images.githubusercontent.com/689204/104832391-99210d80-5845-11eb-8e6d-3b322d87a2de.png width=700>

The top of the generated `json` file looks like this.

```
{"__schema":{"queryType":{"name":"query_root"},"mutationType":{"name":"mutation_root"},"subscriptionType":
{"name":"subscription_root"},"types":[{"kind":"OBJECT","name":"__Directive","fields":[{"name":"args","type":
{"kind":"NON_NULL","ofType":{"kind":"LIST","ofType":{"kind":"NON_NULL","ofType":
```

What am I missing?

Thanks!

**Comment by JoviDeCroock** [comment, marked as answer]
Hey,

TypeScript can't really infer the type it seems, it should be safe to ignore/cast this error, we wmight want to look into how to correctly infer the type from a json file.

The automatic infer will probably be having issues due to us changing the shape of the `introspection` to be optimized (we remove things graphcache doesn't need so you save size)

**Reply by GollyJer** [reply]
Thanks for the response @JoviDeCroock. I'll ignore it for now. 👍

> The automatic infer will probably be having issues due to us changing the shape of the introspection to be optimized (we remove things graphcache doesn't need so you save size)

I tried with the non-minified version and the same thing happens. Just FYI.
